fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/rescheduleloan/data/LoanRescheduleRequestDataValidatorImpl.java

```diff
@@ -200,13 +200,13 @@ public static void validateSupportedParameters(JsonCommand jsonCommand, Set<Stri
     }
 
     public static void validateReschedulingInstallment(DataValidatorBuilder dataValidatorBuilder,
-            LoanRepaymentScheduleInstallment installment) {
+            LoanRepaymentScheduleInstallment installment, LoanScheduleType loanScheduleType) {
         if (installment == null) {
             dataValidatorBuilder.reset().parameter(RescheduleLoansApiConstants.rescheduleFromDateParamName)
                     .failWithCode("repayment.schedule.installment.does.not.exist", "Repayment schedule installment does not exist");
         }
 
-        if (installment != null && installment.isObligationsMet()) {
+        if (installment != null && installment.isObligationsMet() && !LoanScheduleType.PROGRESSIVE.equals(loanScheduleType)) {
```

Comment:
Would you mind to keep the progressive loan related validations in the `ProgressiveLoanRescheduleRequestDataValidator.java` class? Feel free to change the original method from static and you can override the behaviour there.

Reply:
Fixed